lib/screens/task_form_screen.dart

```diff
@@ -16,12 +16,16 @@ class _TaskFormScreenState extends State<TaskFormScreen> {
   late final _task;
   final _formKey = GlobalKey<FormState>();
   late final _formController = TextEditingController(text: _task.title);
+  late final _formDescriptionController =
+      TextEditingController(text: _task.description);
   bool _editing = false;
 
   void _onSubmitPressed(context) {
     if (_formKey.currentState!.validate()) {
       Provider.of<TaskListService>(context, listen: false)
           .upsert(_task..title = _formController.text);
+      Provider.of<TaskListService>(context, listen: false)
+          .upsert(_task..description = _formDescriptionController.text);
```

Comment:
You could reuse the previous provided service + use the cascade operator (..) of Dart like e.g.:
```dart
final taskListService = Provider.of<TaskListService>(context, listen: false);
taskListService.upsert(_task
    ..title = _formController.text
    ..description = _formDescriptionController);
```

* https://dart.dev/guides/language/language-tour#cascade-notation